lib/adhearsion/call_controller.rb

```diff
@@ -222,6 +222,10 @@ def execute_component_and_await_completion(component)
 
       complete_event = component.complete_event
       raise Adhearsion::Error, [complete_event.reason.details, component.inspect].join(": ") if complete_event.reason.is_a? Punchblock::Event::Complete::Error
+      if complete_event.reason.is_a? Punchblock::Event::Complete::Hangup
```

Comment:
Events on call are called asynchronously: the callcontroller must check the result of the complete event synchronously to prevent the execution of his flow. Is there a better way to sync this?